libvalkeycluster/examples/src/example_async.c

```diff
@@ -46,48 +46,48 @@ void disconnectCallback(const redisAsyncContext *ac, int status) {
 
 int main(int argc, char **argv) {
     printf("Connecting...\n");
-    redisClusterAsyncContext *cc =
-        redisClusterAsyncConnect("127.0.0.1:7000", HIRCLUSTER_FLAG_NULL);
+    valkeyClusterAsyncContext *cc =
+        valkeyClusterAsyncConnect("127.0.0.1:7000", VALKEYCLUSTER_FLAG_NULL);
```

Comment:
it's just `0`? Can we rename this to VALKEYCLUSTER_FLAG_NONE or just use 0? "Null" sticks out to me.

Reply:
I think it would be nice to replace the flags used in the Cluster API with a new valkeyClusterOptions,
like its done in standalone:
```
valkeyContext *valkeyConnectWithOptions(const valkeyOptions *options);
```

Then we could set any option with that API, and possibly keep the simpler one for easy usage.
```
valkeyClusterContext *valkeyClusterConnectWithOptions(const valkeyClusterOptions *options);
valkeyClusterContext *valkeyClusterConnect(const char *addrs);
```
With this change the cluster api would look the same as the standalone.